django/db/backends/mysql/base.py

```diff
@@ -253,7 +262,30 @@ def get_connection_params(self):
         # We need the number of potentially affected rows after an
         # "UPDATE", not the number of changed rows.
         kwargs['client_flag'] = CLIENT.FOUND_ROWS
-        kwargs.update(settings_dict['OPTIONS'])
+        # Special handling for transaction isolation level
+        options = copy.copy(settings_dict['OPTIONS'])
+        self.isolation_level = options.pop('isolation_level', None)
+        if self.isolation_level:
+            self.isolation_level = self.isolation_level.lower()
+            if self.isolation_level not in self.isolation_levels:
+                raise ImproperlyConfigured(
+                    "Invalid transaction isolation level '%s' specified.\n"
+                    "Use one of %s or None." % (
+                        self.isolation_level,
+                        ", ".join(repr(str(s)) for s in self.isolation_levels)
+                    ))
+            # We are going to use the var-assignment form of setting transaction isolation
+            # levels, and in that form we say "set tx_isolation='repeatable-read'" rather
+            # than "set transaction isolation level repeatable read"
+            self.isolation_level_value = "'%s'" % self.isolation_level.replace(' ', '-')
+        else:
+            if 'isolation_level' not in settings_dict['OPTIONS']:
```

Comment:
Yes, but IMO the current formatting represents the logic better; the `if` under the `else` is not on an equal footing with the `if`/`else`.